### PR TITLE
SERVER-909 Refactor for circleci/server

### PR DESCRIPTION
**Pull request by nsmith5**
:gear: **Issue**

We've stop being so prescriptive in how folks set up their infrastructure in newer release candidates of server 3.0. That said we still want to provide some helpful Terraform modules to spin up Nomad clients in different cloud providers.

:white_check_mark: **Fix**

- Remove the GKE and EKS modules from this repository as we no longer support them
- Add Nomad AWS module (GCP will follow)
- Add example of using the module

:question: **Tests**

- [x] Example works
- [x] Rendered output of Markdown looks good

### CODE REVIEW

File: shared/nomad-scripts/nomad-startup.sh.tpl [hunk 1]
```diff
@@ -3,24 +3,13 @@
 # One of 'AWS' or 'GCP'. Value passed into template
 export CLOUD_PROVIDER=${cloud_provider}
```
Comment (robinsturmeit) [inline shared/nomad-scripts/nomad-startup.sh.tpl:4]: Hmm… You are removing all the conditionals that deal with GCP/AWS differences, but leave this variable in place and keep the file in the `shared` folder. That strikes me as inconsistent - Either keep the conditionals or move the file to the aws module folder and remove this variable.

**Review by robinsturmeit — CHANGES_REQUESTED** [review]
Apart from the broader discussion about this change we will have later, I had a comment about the Nomad startup script.

**Review by lloydChris — APPROVED** [review]
I agree with @robinsturmeit 's comments, otherwise looks good

File: shared/nomad-scripts/nomad-startup.sh.tpl (hunk 1 above)
Comment (nsmith5) [reply shared/nomad-scripts/nomad-startup.sh.tpl:4]: We'll need to add GCP support shortly. Didn't mean to the support here, this is just a copy and paste of the current state in `circleci/server`. I'd like to keep in `/shared` but do you mind if I add back the GCP specific bits in an iteration?

Comment (robinsturmeit) [reply shared/nomad-scripts/nomad-startup.sh.tpl:4]: Generally no, but experience has taught me that these "I'll add this back in later" kind of things gets lost easily :/

Comment (nsmith5) [reply shared/nomad-scripts/nomad-startup.sh.tpl:4]: Resolved out of band: going to iterate right after this PR with those changes just to unblock the release process.

**Pull request merged**